libuuu/trans.cpp

```diff
@@ -112,26 +113,38 @@ int HIDTrans::read(void *buff, size_t size, size_t *rsize)
 {
 	int ret;
 	int actual;
-	ret = libusb_interrupt_transfer(
-		(libusb_device_handle *)m_devhandle,
-		0x81,
-		(uint8_t*)buff,
-		size,
-		&actual,
-		m_read_timeout
-	);
-
-	*rsize = actual;
 
-	if (ret < 0)
+	for (size_t i = 0; i < size; )
 	{
-		string error;
-		string err;
-		err = "HID(R):";
-		err += libusb_error_name(ret);
-		set_last_err_string(err);
-		return ret;
-	}
+		uint8_t *p = (uint8_t *)buff;
+		p += i;
+		size_t sz;
+		sz = size - i;
+		if (sz > Conf::GetConf().m_USB.m_HIDMaxTransfer)
+			sz = Conf::GetConf().m_USB.m_HIDMaxTransfer;
+
+		ret = libusb_interrupt_transfer(
+				(libusb_device_handle *)m_devhandle,
+				0x81,
+				p,
+				sz,
+				&actual,
+				Conf::GetConf().m_USB.m_HIDTimeout
+				);
```

Comment:
you should update m_read_timeout.

Reply:
read time out seems to be something not tunable if I correctly remember.... only write timeout is meaningfull for tunning (remembered seeing that in libusb documentation...)